dnf5/commands/check/check.cpp

```diff
@@ -0,0 +1,315 @@
+
+    return false;
+}
+
+
+// Finds problems for the installed package defined by Header h.
+// Principle: Adds the installed package to an empty transaction and checks for errors.
+int find_problems(QVA_t qva, rpmts ts, Header h) {
+    int rc = 0;
+
+    if (qva->qva_flags & VERIFY_DEPS) {
+        rpmtsEmpty(ts);
+        (void)rpmtsAddInstallElement(ts, h, NULL, 0, NULL);
+
+        (void)rpmtsCheck(ts);
+        auto te = rpmtsElement(ts, 0);
+        auto ps = rpmteProblems(te);
+        rc = rpmpsNumProblems(ps);
+
+        if (rc > 0) {
+            rpmpsi psi = rpmpsInitIterator(ps);
+            while (auto p = rpmpsiNext(psi)) {
+                auto problem_cstr = rpmProblemGetStr(p);
+                libdnf_assert(problem_cstr != NULL, "command \"check\": rpmProblemGetStr() returns NULL");
+                std::string_view problem_str = problem_cstr;
+                auto cnevr = rpmProblemGetPkgNEVR(p);
+                libdnf_assert(cnevr != NULL, "command \"check\": rpmProblemGetPkgNEVR() returns NULL");
+                std::string_view nevr = cnevr;
+                auto alt_cnevr = rpmProblemGetAltNEVR(p);
+                libdnf_assert(alt_cnevr != NULL, "command \"check\": rpmProblemGetAltNEVR() returns NULL");
+                std::string_view alt_nevr = alt_cnevr;
+
+                switch (rpmProblemGetType(p)) {
+                    case RPMPROB_REQUIRES:
+                        if (all || dependencies) {
+                            problems[std::string{alt_nevr}].insert(Problem{
+                                .type = ProblemType::MISSING_REQUIRE,
+                                .nevra = "",
+                                .file_or_provide = std::string{problem_str}});
+                        }
+                        break;
+                    case RPMPROB_CONFLICT:
+                        if (all || dependencies) {
+                            auto nevr_compare = nevr.compare(alt_nevr);
+                            if (nevr_compare == 0) {
+                                // skip self conflicts
+                                break;
+                            }
+                            problems[std::string{nevr}].insert(Problem{
+                                .type = ProblemType::CONFLICT,
+                                .nevra = std::string{alt_nevr},
+                                .file_or_provide = std::string{problem_str}});
+                        }
+                        break;
+                    case RPMPROB_OBSOLETES:
+                        if (all || obsoleted) {
+                            problems[std::string{problem_str}].insert(Problem{
+                                .type = ProblemType::OBSOLETED, .nevra = std::string{alt_nevr}, .file_or_provide = ""});
+                        }
+                        break;
+                    default:;
+                        //std::cout << fmt::format("{} problem {}: {}", nevr, problem_str, alt_nevr) << std::endl;
```

Comment:
May I ask you for the purpose of the comment?

Reply:
The remainder after debugging. I deleted it.